doc/releases/changelog-dev.md

```diff
@@ -116,6 +116,11 @@
 
 <h4>Capturing and representing hybrid programs</h4>
 
+* Execution with capture enabled now follows a new execution pipeline and natively passes the
+  captured jaxpr to the device. Since it no longer falls back to the old pipeline, execution
+  only works with a reduced feature set.
```

Comment:
Is there a comprehensive overview of things that stop working? I noticed
- non-device shots
- non-backprop differentiation
- terminal MCM returns
- Some postselection-related behaviour (got a bit confused there, sorry).
Might be nice to provide an overview (even if not complete).

Reply:
This would be very good to document here, even if incomplete!

Reply:
Added some documentation to the top of the `_capture_qnode.py` file.  While non-exhaustive, it hopefully covers the major features and blockers.